internal/datacoord/segment_manager.go

```diff
@@ -237,7 +271,8 @@ func (s *SegmentManager) AllocSegment(ctx context.Context, collectionID UniqueID
 		if err != nil {
 			return
 		}
-		success, err = s.alloc(status, requestRows)
+		info, err = s.meta.GetSegment(status.id)
```

Comment:
The error returned by GetSegment is ignored. Maybe we can extract `s.meta.GetSegment` and the error check and log print to a single method.